modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/defragmentation/CachePartitionDefragmentationManager.java

```diff
@@ -1072,61 +1114,76 @@ public synchronized void onCacheGroupFinish(CacheGroupContext grpCtx) {
         }
 
         /** */
-        public synchronized void onCacheGroupSkipped(CacheGroupContext grpCtx) {
+        private synchronized void onCacheGroupSkipped(CacheGroupContext grpCtx, int partitions) {
             scheduledGroups.remove(grpCtx.cacheOrGroupName());
 
             skippedGroups.add(grpCtx.cacheOrGroupName());
+
+            defragmentedPartitionCount += partitions;
         }
 
         /** */
-        public synchronized void onFinish() {
+        private synchronized void onFinish() {
             finishTs = System.currentTimeMillis();
 
             log.info("Defragmentation process completed. Time: " + (finishTs - startTs) * 1e-3 + "s.");
         }
 
-        /** {@inheritDoc} */
-        @Override public synchronized String toString() {
-            StringBuilder sb = new StringBuilder();
-
-            if (!finishedGroups.isEmpty()) {
-                sb.append("Defragmentation is completed for cache groups:\n");
+        /** */
+        private synchronized Status copy() {
```

Comment:
So, who calls it? All I see is this:

public Status status() {
    return status;
}